util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/SearchBackendContainer.java

```diff
@@ -0,0 +1,192 @@
+		if ( !SEARCH_CONTAINER.isRunning() ) {
+			synchronized (SEARCH_CONTAINER) {
+				if ( !SEARCH_CONTAINER.isRunning() ) {
+					SEARCH_CONTAINER.start();
+				}
+			}
+		}
+	}
+
+	private static GenericContainer<?> elasticsearch(DockerImageName dockerImageName) {
+		GenericContainer<?> container = common( dockerImageName )
+				.withEnv( "logger.level", "WARN" )
+				.withEnv( "discovery.type", "single-node" )
+				// Older images require HTTP authentication for all requests;
+				// it's not practical for testing, so we disable that:
+				.withEnv( "xpack.security.enabled", "false" )
+				// Limit the RAM usage.
+				// Recent versions of ES limit themselves to 50% of the total available RAM,
+				// but on CI this is sometimes too much, as we also have the Maven JVM
+				// and the JVM that runs tests taking up a significant amount of RAM,
+				// leaving too little for filesystem caches and resulting in freezes:
+				.withEnv( "ES_JAVA_OPTS", "-Xms1g -Xmx1g" )
+				// Disable disk-based shard allocation thresholds: on large, relatively full disks (>90% used),
+				// it will lead to index creation to get stuck waiting for other nodes to join the cluster,
+				// which will never happen since we only have one node.
+				// See https://www.elastic.co/guide/en/elasticsearch/reference/7.17/modules-cluster.html#disk-based-shard-allocation
+				.withEnv( "cluster.routing.allocation.disk.threshold_enabled", "false" );
+
+		ElasticsearchVersion version =
+				ElasticsearchVersion.of( ElasticsearchDistributionName.ELASTIC, dockerImageName.getVersionPart() );
+
+		if ( version.majorOptional().orElse( Integer.MIN_VALUE ) == 8 ) {
+			if ( version.minor().orElse( Integer.MAX_VALUE ) > 7 ) {
+				container.withEnv( "cluster.deprecation_indexing.enabled", "false" )
+						.withEnv( "xpack.profiling.enabled", "false" )
+						.withEnv( "xpack.ent_search.enabled", "false" );
+			}
+			return container.withEnv( "cluster.deprecation_indexing.enabled", "false" )
+					.withEnv( "indices.lifecycle.history_index_enabled", "false" )
+					.withEnv( "slm.history_index_enabled", "false" )
+					.withEnv( "stack.templates.enabled", "false" )
+					.withEnv( "xpack.ml.enabled", "false" )
+					.withEnv( "xpack.monitoring.templates.enabled", "false" )
+					.withEnv( "xpack.watcher.enabled", "false" );
+		}
+
+		if ( version.majorOptional().orElse( Integer.MIN_VALUE ) == 7 ) {
+			if ( version.minor().orElse( Integer.MAX_VALUE ) > 15 ) {
+				container.withEnv( "cluster.deprecation_indexing.enabled", "false" )
+						.withEnv( "slm.history_index_enabled", "false" );
+			}
+			if ( version.minor().orElse( Integer.MAX_VALUE ) > 8 ) {
+				container.withEnv( "stack.templates.enabled", "false" );
+			}
+			if ( version.minor().orElse( Integer.MAX_VALUE ) > 5 ) {
+				container.withEnv( "indices.lifecycle.history_index_enabled", "false" );
+			}
+			return container.withEnv( "xpack.ml.enabled", "false" )
+					.withEnv( "xpack.watcher.enabled", "false" );
```

Comment:
We don't test against ES 7.9 or lower anymore:

```suggestion
			return container
					.withEnv( "indices.lifecycle.history_index_enabled", "false" )
					.withEnv( "stack.templates.enabled", "false" )
					.withEnv( "xpack.ml.enabled", "false" )
					.withEnv( "xpack.watcher.enabled", "false" );
```